includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php

```diff
@@ -1381,6 +1365,22 @@ public function add_contact( $contact, $list_id = false ) {
 				$payload[ 'field[%' . $field_perstag . '%,0]' ] = (string) $value; // Per ESP documentation, "leave 0 as is".
 			}
 		}
+
+		$contact_data          = $this->get_contact_data( $email );
+		$existing_email        = isset( $contact['existing_contact_data']['email'] ) ? $contact['existing_contact_data']['email'] : '';
+		$existing_contact_data = $this->get_contact_data( $existing_email );
+		if ( ! is_wp_error( $contact_data ) || ! is_wp_error( $existing_contact_data ) ) {
+			$action               = 'contact_edit';
+			$payload['id']        = is_wp_error( $contact_data ) ? $existing_contact_data['id'] : $contact_data['id'];
+			$payload['overwrite'] = 0;
+			// If the email address exists, but is different from the one we're trying to upsert, delete the existing contact.
+			if ( ! is_wp_error( $contact_data ) && ! is_wp_error( $existing_contact_data ) && $existing_email !== $email ) {
+				$delete_res = $this->delete_contact( $existing_email );
+				if ( is_wp_error( $delete_res ) ) {
+					Newspack_Newsletters_Logger::log( 'Error deleting existing contact during upsert: ' . $delete_res->get_error_message() );
+				}
+			}
```

Comment:
ActiveCampaign's automations can rely a lot on contact activity and history. Won't this introduce a reset and impact how publishers handle their campaigns?

Reply:
I think it would, although I don't think there is an alternative solution since there is no way to "replace" a contacts activity and history with another's that I know of. This is also something we don't handle currently and is probably an edge case, so I don't think its something we need to worry about unless we get a report.